src/app/controllers/locations/list.js

```diff
@@ -0,0 +1,115 @@
+            //  the view before changing the location path. Is there a
+            //  better place to put this? Perhaps something in the routing
+            //  to intercept the route and direct traffic by permission?
+            if (!(authService.isAdministrator() ||
+                  authService.isCoordinator())) {
+                $log.warn('not authorized to access this path');
+                $location.path('/');
+                return;
+            }
+
+            $scope.loading = true;
+
+            flnd.locationList.retrieveRecords(config, $scope, messenger, authService);
+
+            $scope.gridOptions = {
+                enableFiltering: true,
+                rowTemplate: 'app/views/row.html',
+                onRegisterApi: function (gridApi) {
+                    $scope.gridApi = gridApi;
+                },
+                data: 'list',
+                columnDefs: [
+                    { field: 'description', displayName: 'Description' },
+                    { field: 'latitude', displayName: 'Latitude' },
+                    { field: 'longitude', displayName: 'Longitude' },
+                    {
+                        field: 'id',
+                        displayName: '',
+                        cellTemplate: '\
+                        <div class="ui-grid-cell-contents" title="Edit">\
+                          <button class="btn btn-primary btn-xs" ng-click="grid.appScope.editLocation(\'{{row.entity.id}}\')" \
+                           aria-label="edit">\
+                              <span class="glyphicon glyphicon-pencil" aria-hidden="true"></span>\
+                          </button>\
+                        </div>',
+                        enableFiltering: false,
+                        width: '10%',
+                        enableColumnMenu: false
+                    }
+                ]
+            };
+
+            $scope.createLocation = function () {
+                var modal = $uibModal.open({
+                    animation: true,
+                    templateUrl: '/app/views/locations/create.html',
+                    controller: 'LocationCreateController',
+                    size: 'lg'
+                });
+                modal.result.then(function ok() {
+                    // Re-load the grid
+                    flnd.locationList.retrieveRecords(config, $scope, messenger, authService);
+                }, function dismissed() {
+                    // no action required
+                });
+            };
+
+            $scope.editLocation = function(id) {
+                var modal = $uibModal.open({
+                    animation: true,
+                    templateUrl: '/app/views/locations/edit.html',
+                    controller: 'LocationEditController',
+                    size: 'lg',
+                    resolve: {
+                        id: function() {
+                            return id;
+                        }
```

Comment:
This is how you pass a parameter to a modal popup's controller. It *must* be a function - you can't use `id: id` because UI Grid will then attempt to execute the value of `id`, which is clearly nonsensical.